support time-based status format

I suggest a method to support a continuous output during deck operation that includes both timecode, the standby flag, cassette out flag, local flag, the status of byte 1 (play, eject, stop, rew, ff, rec, play) and add a new line to the output when changed.

I'd like a method to automatically stop the output so perhaps it could be set to a timer (`-t 90` to read and output for 90 seconds) or an option to stop the output when `stop=1` in byte 1 of the status output.

This is a good start. 

When I build with qt 5.15.8 it's fine, but with qt 6.3.2 I get:

```
$ make
/Library/Developer/CommandLineTools/usr/bin/clang++ -c -pipe -stdlib=libc++ -O2 -std=gnu++1z  -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -mmacosx-version-min=11 -Wall -Wextra -fPIC -DQT_NO_DEBUG -DQT_GUI_LIB -DQT_SERIALPORT_LIB -DQT_CORE_LIB -I. -I. -I. -I/usr/local/lib/QtGui.framework/Headers -I/usr/local/lib/QtSerialPort.framework/Headers -I/usr/local/lib/QtCore.framework/Headers -I. -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/OpenGL.framework/Headers -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/AGL.framework/Headers -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/OpenGL.framework/Headers -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/AGL.framework/Headers -I/usr/local/share/qt/mkspecs/macx-clang -F/usr/local/lib -o sony9pin.o sony9pin.cpp
sony9pin.cpp:686:38: error: use of undeclared identifier 'QRegExp'
        QStringList tc = param.split(QRegExp(":|;"));
                                     ^
1 error generated.
make: *** [sony9pin.o] Error 1
```
Then I can run:

```
$ ./sony9pin.app/Contents/MacOS/sony9pin -c 4 0 
Info: continous mode.
<Remote Status>
==================
------byte 0------
Cassette Out :  0
Servo Ref Mis:  0
Local        :  0
------byte 1------
Standby      :  1
Stop         :  0
Eject        :  0
Rewind       :  0
Forward      :  0
Record       :  0
Play         :  1
------byte 2------
Servo Lock   :  1
TSO Mode     :  0
Shuttle      :  0
Jog          :  0
Var          :  0
Direction    :  0
Still        :  0
Cue Up       :  0
------byte 3------
Auto Mode    :  0
Freeze On    :  0
CF Mode      :  0
Audio Out Set:  0
Audio In Set :  0
Out Set      :  0
In Set       :  0
------byte 4------
Select EE    :  0
Full EE      :  0
Edit         :  0
Review       :  0
Auto Edit    :  0
Preview      :  0
Preroll      :  0
------byte 5------
Insert       :  0
Assemble     :  0
Video        :  0
A4           :  0
A3           :  0
A2           :  0
A1           :  0
------byte 6------
Lamp Still   :  0
Lamp Fwd     :  0
Lamp Rev     :  0
SRCH Led 8   :  0
SRCH Led 4   :  0
SRCH Led 2   :  0
SRCH Led 1   :  0
------byte 7------
AUD Split    :  0
Sync Act     :  0
Spot Erase   :  0
In Out       :  0
------byte 8------
Buzzer       :  0
Lost Lock    :  0
Near EOT     :  0
EOT          :  0
CF Lock      :  0
SVO Alarm    :  0
Sys Alarm    :  0
Rec Inhibit  :  1
------byte 9------
FNC Abort    :  0
==================
2023-02-03T14:49:42.803 00:00:01;00 cassette_out=0 servo_ref_missing=0 local=0 standby=1 stop=0 eject=0 rewind=0 forward=0 record=0 play=1 servo_lock=1 tso_mode=0 shuttle=0 jog=0 var=0 direction=0 still=0 cue_up=0 lamp_still=0 lamp_fwd=0 lamp_rev=0 near_eot=0 eot=0 cf_lock=0 svo_alarm=0 sys_alarm=0 rec_inhib=1
2023-02-03T14:50:30.377 00:00:01;01 local=1
2023-02-03T14:50:34.933 00:00:01;00 local=0
2023-02-03T14:50:35.732 00:00:01;01 local=1
2023-02-03T14:50:38.418 00:04:00;01 play=0 servo_lock=0 direction=1
2023-02-03T14:50:38.769 00:00:01;01 play=1 direction=0
2023-02-03T14:50:39.105 servo_lock=1
2023-02-03T14:50:40.272 00:04:00;01 play=0 servo_lock=0 direction=1
2023-02-03T14:50:43.949 00:00:01;01 play=1 servo_lock=1 direction=0
2023-02-03T14:50:43.981 servo_lock=0
2023-02-03T14:50:45.915 servo_lock=1
2023-02-03T14:50:52.341 00:00:01;00 local=0
2023-02-03T14:50:53.684 00:00:01;01 local=1
2023-02-03T14:50:54.228 00:00:01;00 local=0
2023-02-03T14:50:58.128 00:00:01;01 local=1
2023-02-03T14:50:58.416 servo_lock=0
2023-02-03T14:50:58.432 00:02:00;01 play=0 still=1
2023-02-03T14:51:02.157 00:02:00;00 local=0
2023-02-03T14:51:28.246 00:00:20;00 standby=0 stop=1 still=0
2023-02-03T14:53:32.935 00:00:20;01 local=1
2023-02-03T14:53:33.782 00:00:01;01 standby=1 stop=0 play=1 servo_lock=1
2023-02-03T14:53:33.814 servo_lock=0
2023-02-03T14:53:36.020 servo_lock=1
2023-02-03T14:53:46.906 00:00:00;01 play=0 servo_lock=0
2023-02-03T14:53:52.518 00:00:01;01 play=1 servo_lock=1
2023-02-03T14:53:52.566 servo_lock=0
2023-02-03T14:53:52.965 servo_lock=1
2023-02-03T14:53:56.882 servo_lock=0
2023-02-03T14:53:56.898 00:00:00;01 play=0
2023-02-03T14:54:01.837 00:00:01;01 play=1
2023-02-03T14:54:02.253 servo_lock=1
2023-02-03T14:54:06.601 00:00:00;01 play=0 servo_lock=0
2023-02-03T14:54:13.348 servo_lock=1
2023-02-03T14:54:13.364 00:00:01;01 play=1
2023-02-03T14:54:13.396 servo_lock=0
2023-02-03T14:54:13.795 servo_lock=1
2023-02-03T14:54:14.019 00:00:00;01 play=0 servo_lock=0
2023-02-03T14:55:24.932 servo_lock=1
2023-02-03T14:55:24.948 00:00:01;01 play=1
2023-02-03T14:55:24.996 servo_lock=0
2023-02-03T14:55:25.396 servo_lock=1
2023-02-03T14:55:30.975 servo_lock=0
2023-02-03T14:55:30.991 00:00:00;01 play=0
2023-02-03T14:55:38.952 00:00:01;01 play=1 servo_lock=1
2023-02-03T14:55:39.016 servo_lock=0
2023-02-03T14:55:39.287 00:04:00;01 play=0 direction=1
2023-02-03T14:55:42.900 servo_lock=1
2023-02-03T14:55:42.916 00:00:01;01 play=1 direction=0
2023-02-03T14:55:42.948 servo_lock=0
2023-02-03T14:55:44.882 servo_lock=1
2023-02-03T14:56:02.658 servo_lock=0
2023-02-03T14:56:02.674 00:04:00;01 play=0 direction=1
2023-02-03T14:56:12.777 00:00:01;01 play=1 direction=0
2023-02-03T14:56:14.711 servo_lock=1
```

There's something off about the timecodes. Rather than match the timecode on the tape, they're composed of 0's, 1's, 2's, and 4's. This seems to report a line whenever the status flags changed, but I'd also like a version where it reports whenever the timecode changes, feasible? Also when the timecode is unavailable, can you leave a --:--:--:-- in place?

--- a/tools/sony9pin/sony9pin.cpp
+++ b/tools/sony9pin/sony9pin.cpp
@@ -5,25 +5,46 @@
  */
 
 #include <QCoreApplication>
+#include <QRegularExpression>
 #include <QSerialPort>
 #include <QSerialPortInfo>
+#include <QDateTime>
 #include <QThread>
 #include <iomanip>
 #include <iostream>
 #include <iterator>
+#include <sstream>
+
 using namespace std;
 
 const char* version = "1.0";
 
 // #define SONY9PINREMOTE_DEBUGLOG_ENABLE
 #include "Sony9PinRemote/Sony9PinRemote.h"
+
+bool operator!=(const sony9pin::TimeCode& first, const sony9pin::TimeCode& second) {
+  return first.is_cf != second.is_cf ||
+         first.is_df != second.is_df ||
+         first.frame != second.frame ||
+         first.second != second.second ||
+         first.minute != second.minute ||
+         first.hour != second.hour;
+}
+
+struct State {
+  Sony9PinRemote::TimeCode tc;
+  Sony9PinRemote::Status st;
+};
+
 Sony9PinRemote::Controller deck;
 QSerialPort serialPort;
+State lastState;
 
 extern map<uint16_t, pair<string, vector<string>>> devices;
 
 void options(const char* const prefix = "") {
   std::cerr << prefix << "Options:\n"
+    << prefix << "-c, --continuous: report deck state until stop bit is set\n"
     << prefix << "-v, --verbose: verbose mode\n"
     << prefix << "-V, --version: show version\n"
     << prefix << "-h, --help: show help\n"
@@ -520,7 +541,7 @@ int main(int argc, char* argv[]) {
   if (!argumentList.isEmpty())
     commandName = argumentList.takeFirst();
 
-  bool verbose = false;
+  bool verbose = false, continuous = false;
   while (!argumentList.isEmpty())
   {
     if (argumentList.first() == "--help" || argumentList.first() == "-h") {
@@ -534,6 +555,11 @@ int main(int argc, char* argv[]) {
     else if (argumentList.first() == "--verbose" || argumentList.first() == "-v") {
         verbose = true;
         cerr << "Info: verbose mode.\n";
+        argumentList.removeFirst();
+    }
+    else if (argumentList.first() == "--continuous" || argumentList.first() == "-c") {
+        continuous = true;
+        cerr << "Info: continuous mode.\n";
         argumentList.removeFirst();
     }
     else
@@ -551,7 +577,7 @@ int main(int argc, char* argv[]) {
   }
 
   auto is_interactive = false;
-  if (argumentList.isEmpty()) {
+  if (!continuous && argumentList.isEmpty()) {
     interactive(is_interactive);
   }
   while (!argumentList.isEmpty() || is_interactive) {
@@ -569,7 +595,10 @@ int main(int argc, char* argv[]) {
     }
     switch (value) {
       case '-': {
-        interactive(is_interactive);
+        if (!continuous)
+            interactive(is_interactive);
+          else
+            cerr << "Error: interactive input unavaiable in continuous mode.\n";
         break;
       }
       case '0': {
@@ -658,7 +687,7 @@ int main(int argc, char* argv[]) {
           param = argumentList.takeFirst();
         }
 
-        QStringList tc = param.split(QRegExp(":|;"));
+        QStringList tc = param.split(QRegularExpression("[:;]"));
         if (tc.size() != 4) {
           cerr << "Error: invalid timecode " << param.toStdString() << ".\n";
           return 1;
@@ -697,6 +726,191 @@ int main(int argc, char* argv[]) {
       default: {
         std::cerr << "Error: unknown command " << value << ".\n ";
       }
+    }
+  }
+
+  if (continuous) {
+    bool first=true;
+    bool stop = false;
+    while (!stop) {
+      State state;
+      bool print = false;
+
+      deck.status_sense();
+      if (!deck.parse_until(1000))
+        std::cerr << "Error: parse failed.\n";
+
+      if (!test_ack()) {
+        std::cout << "Info: parse issue.\n";
+        deck.print_nak();
+      }
+      state.st = deck.status();
+
+      deck.current_time_sense_timer1();
+      if (!deck.parse_until(1000))
+        std::cerr << "Error: parse failed.\n";
+
+      if (!test_ack()) {
+        std::cout << "Info: parse issue.\n";
+        deck.print_nak();
+      }
+      state.tc = deck.timecode();
+
+      std::stringstream ss;
+      ss << dec << ' '
+         << setw(2) << setfill('0') << (unsigned int)state.tc.hour << ':'
+         << setw(2) << setfill('0') << (unsigned int)state.tc.minute << ':'
+         << setw(2) << setfill('0') << (unsigned int)state.tc.second << ';'
+         << setw(2) << setfill('0') << (unsigned int)state.tc.frame
+         << resetiosflags(std::ios::dec);
+
+      if (first || state.tc != lastState.tc) {
+        print = true;
+      }
+
+      if (first || state.st.b_cassette_out != lastState.st.b_cassette_out) {
+        ss << " cassette_out=" << (unsigned int)state.st.b_cassette_out;
+        print = true;
+      }
+
+      if (first || state.st.b_servo_ref_missing != lastState.st.b_servo_ref_missing) {
+        ss << " servo_ref_missing=" << (unsigned int)state.st.b_servo_ref_missing;
+        print = true;
+      }
+
+      if (first || state.st.b_local != lastState.st.b_local) {
+        ss << " local=" << (unsigned int)state.st.b_local;
+        print = true;
+      }
+
+      if (first || state.st.b_standby != lastState.st.b_standby) {
+        ss << " standby=" << (unsigned int)state.st.b_standby;
+        print = true;
+      }
+
+      if (first || state.st.b_stop != lastState.st.b_stop) {
+        ss << " stop=" << (unsigned int)state.st.b_stop;
+        print = true;
+
+        if (state.st.b_stop)
+          stop = true;
+      }
+
+      if (first || state.st.b_eject != lastState.st.b_eject) {
+        ss << " eject=" << (unsigned int)state.st.b_eject;
+        print = true;
+      }
+
+      if (first || state.st.b_rewind != lastState.st.b_rewind) {
+        ss << " rewind=" << (unsigned int)state.st.b_rewind;
+        print = true;
+      }
+
+      if (first || state.st.b_forward != lastState.st.b_forward) {
+        ss << " forward=" << (unsigned int)state.st.b_forward;
+        print = true;
+      }
+
+      if (first || state.st.b_record != lastState.st.b_record) {
+        ss << " record=" << (unsigned int)state.st.b_record;
+        print = true;
+      }
+
+      if (first || state.st.b_play != lastState.st.b_play) {
+        ss << " play=" << (unsigned int)state.st.b_play;
+        print = true;
+      }
+
+      if (first || state.st.b_servo_lock != lastState.st.b_servo_lock) {
+        ss << " servo_lock=" << (unsigned int)state.st.b_servo_lock;
+        print = true;
+      }
+
+      if (first || state.st.b_tso_mode != lastState.st.b_tso_mode) {
+        ss << " tso_mode=" << (unsigned int)state.st.b_tso_mode;
+        print = true;
+      }
+
+      if (first || state.st.b_shuttle != lastState.st.b_shuttle) {
+        ss << " shuttle=" << (unsigned int)state.st.b_shuttle;
+        print = true;
+      }
+
+      if (first || state.st.b_jog != lastState.st.b_jog) {
+        ss << " jog=" << (unsigned int)state.st.b_jog;
+        print = true;
+      }
+
+      if (first || state.st.b_var != lastState.st.b_var) {
+        ss << " var=" << (unsigned int)state.st.b_var;
+        print = true;
+      }
+
+      if (first || state.st.b_direction != lastState.st.b_direction) {
+        ss << " direction=" << (unsigned int)state.st.b_direction;
+        print = true;
+      }
+
+      if (first || state.st.b_still != lastState.st.b_still) {
+        ss << " still=" << (unsigned int)state.st.b_still;
+        print = true;
+      }
+
+      if (first || state.st.b_cue_up != lastState.st.b_cue_up) {
+        ss << " cue_up=" << (unsigned int)state.st.b_cue_up;
+        print = true;
+      }
+
+      if (first || state.st.b_lamp_still != lastState.st.b_lamp_still) {
+        ss << " lamp_still=" << (unsigned int)state.st.b_lamp_still;
+        print = true;
+      }
+
+      if (first || state.st.b_lamp_fwd != lastState.st.b_lamp_fwd) {
+        ss << " lamp_fwd=" << (unsigned int)state.st.b_lamp_fwd;
+        print = true;
+      }
+
+      if (first || state.st.b_lamp_rev != lastState.st.b_lamp_rev) {
+        ss << " lamp_rev=" << (unsigned int)state.st.b_lamp_rev;
+        print = true;
+      }
+
+      if (first || state.st.b_near_eot != lastState.st.b_near_eot) {
+        ss << " near_eot=" << (unsigned int)state.st.b_near_eot;
+        print = true;
+      }
+
+      if (first || state.st.b_eot != lastState.st.b_eot) {
+        ss << " eot=" << (unsigned int)state.st.b_eot;
+        print = true;
+      }
+
+      if (first || state.st.b_cf_lock != lastState.st.b_cf_lock) {
+        ss << " cf_lock=" << (unsigned int)state.st.b_cf_lock;
+        print = true;
+      }
+
+      if (first || state.st.b_svo_alarm != lastState.st.b_svo_alarm) {
+        ss << " svo_alarm=" << (unsigned int)state.st.b_svo_alarm;
+        print = true;
+      }
+
+      if (first || state.st.b_sys_alarm != lastState.st.b_sys_alarm) {
+        ss << " sys_alarm=" << (unsigned int)state.st.b_sys_alarm;
+        print = true;
+      }
+
+      if (first || state.st.b_rec_inhib != lastState.st.b_rec_inhib) {
+        ss << " rec_inhib=" << (unsigned int)state.st.b_rec_inhib;
+        print = true;
+      }
+
+      if (print) {
+        cout << QDateTime::currentDateTime().toString(Qt::ISODateWithMs).toStdString() << ss.str() << '\n';
+        lastState=state;
+      }
+      first=false;
     }
   }
 

